crates/worker/src/partition/snapshots/repository.rs

```diff
@@ -414,41 +415,44 @@ impl SnapshotRepository {
                 .child(partition_id.to_string())
                 .child(latest.path.as_str())
                 .child(filename);
-            let file_path = snapshot_dir.path().join(filename);
+            let local_path = snapshot_dir.path().join(filename);
             let concurrency_limiter = Arc::clone(&concurrency_limiter);
             let object_store = Arc::clone(&self.object_store);
+            let snapshot_id = snapshot_metadata.snapshot_id;
+            let snapshot_filename = filename.to_owned();
 
             let handle = downloads.build_task().name(filename).spawn(async move {
                 let _permit = concurrency_limiter.acquire().await?;
-                debug!(%key, "Downloading snapshot file");
+                debug!(%key, "Downloading snapshot object");
                 let mut file_data = StreamReader::new(
                     object_store
                         .get(&key)
                         .await
-                        .map_err(|e| anyhow!("Failed to download snapshot file {:?}: {}", key, e))?
+                        .map_err(|e| anyhow!("Failed to download partition {partition_id} snapshot {snapshot_id} file {key:?}: {e}"))?
                         .into_stream(),
                 );
 
                 let mut snapshot_file =
-                    tokio::fs::File::create_new(&file_path).await.map_err(|e| {
-                        anyhow!("Failed to create snapshot file {:?}: {}", file_path, e)
+                    tokio::fs::File::create_new(&local_path).await.map_err(|e| {
+                        anyhow!("Failed to create local partition {partition_id} snapshot file {local_path:?}: {e}")
                     })?;
                 let size = io::copy(&mut file_data, &mut snapshot_file)
                     .await
-                    .map_err(|e| anyhow!("Failed to download snapshot file {:?}: {}", key, e))?;
+                    .map_err(|e| anyhow!("Failed to download snapshot object {:?}: {}", key, e))?;
                 snapshot_file.shutdown().await?;
 
                 if size != expected_size as u64 {
                     return Err(anyhow!(
-                        "Downloaded snapshot file {:?} has unexpected size: expected {}, got {}",
-                        key,
+                        "Downloaded partition {partition_id} snapshot {snapshot_id} component file {:?} \
+                        has unexpected size: expected: {}, actual: {}",
+                        snapshot_filename,
                         expected_size,
                         size
                     ));
                 }
-                debug!(%key, ?size, "Downloaded snapshot data file to {:?}", file_path);
+                debug!(%key, ?size, "Downloaded snapshot object {}", local_path.display());
                 anyhow::Ok(())
-            })?;
+            }.instrument(debug_span!(parent: &Span::current(), "download_object", file = %filename)))?;
```

Comment:
I don't think you need to set the parent span explicitly here, do you?

Reply:
No! Thought it was needed to get the nesting right - but not so.